multi-agent/lib/mas/resources/service.py

```
@@ -297,35 +384,226 @@ def validate_config(
+        available_to_all: bool = None,
+    ) -> Resource:
+        """Update a built-in resource (admin only)."""
+        return self.builtin.update_builtin(
+            rid, config=config, name=name, available_to_all=available_to_all,
+        )
+
+    def update_builtin_with_cascade(
+        self,
+        rid: str,
+        *,
+        config: dict = None,
```

Ruff (0.15.21)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)